### ISSUE TITLE
Dog Leap causes crash with multiple WithInfantryBodies

### ISSUE THREAD

**Issue by CH4Code** [OpenRA/OpenRA#14554]
Hi wanted to give the Dog multiple bodies, one for water one for land which worked fine on other infantry actors (tanya) but this caused a crash on the dog acctor type.

In `Leap` which gets called by `AttackLeap` probably this line causes it:
```
// HACK: why isn't this using the interface?
self.Trait<WithInfantryBody>().Attacking(self, Target.FromActor(target), a);
```
since I must disable one (the first) `WithInfantryBody `with `RequiresPrerequisite` while the dog is in water.

**Comment by CH4Code** [comment]
OpenRA engine version release-20171014
Red Alert mod version release-20171014
on map 8380f72e70ab01b9622c817d4377eb51d3921275 (Sidestep - DEV Underused Options Empowerment (UOE) by SoScared - SirCake).
Date: 2017-12-21 12:35:22Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Actor dog has multiple traits of type `OpenRA.Mods.Common.Traits.Render.WithInfantryBody`
   bei OpenRA.TraitDictionary.TraitContainer`1.GetOrDefault(UInt32 actor)
   bei OpenRA.TraitDictionary.TraitContainer`1.Get(UInt32 actor)
   bei OpenRA.TraitDictionary.Get[T](Actor actor)
   bei OpenRA.Actor.Trait[T]()
   bei OpenRA.Mods.Cnc.Activities.Leap..ctor(Actor self, Actor target, Armament a, WDist speed, WAngle angle)
   bei OpenRA.Mods.Cnc.Traits.AttackLeap.DoAttack(Actor self, Target target, IEnumerable`1 armaments)
   bei OpenRA.Mods.Common.Activities.Attack.InnerTick(Actor self, AttackBase attack)
   bei OpenRA.Mods.Common.Activities.Attack.Tick(Actor self)
   bei OpenRA.Activities.Activity.TickOuter(Actor self)
   bei OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act)
   bei OpenRA.Actor.Tick()
   bei OpenRA.World.Tick()
   bei OpenRA.Game.InnerLogicTick(OrderManager orderManager)
   bei OpenRA.Game.LogicTick()
   bei OpenRA.Game.Loop()
   bei OpenRA.Game.Run()
   bei OpenRA.Program.Run(String[] args)
   bei OpenRA.Program.Main(String[] args)

**Comment by CH4Code** [comment]
```
DOG:
	Mobile:
		TerrainSpeeds:
			Water: 60
	GrantConditionOnTerrain:
		TerrainTypes: Water
		Condition: swimming
	Targetable:
		RequiresCondition: !parachute && !swimming
	Targetable@SEA:
		TargetTypes: Ground, Water, Infantry
		RequiresCondition: !parachute && swimming
	WithInfantryBody:
		RequiresCondition: !swimming
	WithInfantryBody@SEA:
		MoveSequence: swim
		DefaultAttackSequence: swim_shoot
		IdleSequences: swim_idle
		StandSequences: swim_stand
		RequiresCondition: swimming
	WithDeathAnimation:
		RequiresCondition: !swimming
```

**Comment by abcdefg30** [comment]
This has been fixed, dogs use multiple WIBs since #15008.
